scipy/interpolate/tests/test_rgi.py

```diff
@@ -484,6 +482,67 @@ def test_fill_value(self, method):
                                          method=method, bounds_error=False)
         assert np.isnan(interp([10]))
 
+    @parametrize_rgi_interp_methods
+    def test_nonscalar_values(self, method):
+        # Verify that non-scalar valued values also works
+        points = [(0.0, 0.5, 1.0, 1.5, 2.0, 2.5)] * 2 + [
+            (0.0, 5.0, 10.0, 15.0, 20, 25.0)
+        ] * 2
+
+        rng = np.random.default_rng(1234)
+        values = rng.random((6, 6, 6, 6, 8))
+        sample = rng.random((7, 11, 4))
+
+        interp = RegularGridInterpolator(points, values, method=method,
+                                         bounds_error=False)
+        v = interp(sample)
+        assert_equal(v.shape, (7, 11, 8), err_msg=method)
+
+        vs = []
+        for j in range(8):
+            interp = RegularGridInterpolator(points, values[..., j],
+                                             method=method,
+                                             bounds_error=False)
+            vs.append(interp(sample))
+        v2 = np.array(vs).transpose(1, 2, 0)
+
+        assert_allclose(v, v2, atol=1e-14, err_msg=method)
+
+    @parametrize_rgi_interp_methods
+    def test_nonscalar_values_2(self, method):
+        # Verify that non-scalar valued values also work : use different
+        # lengths of axes to simplify tracing the internals
+        points = [(0.0, 0.5, 1.0, 1.5, 2.0, 2.5),
+                  (0.0, 0.5, 1.0, 1.5, 2.0, 2.5, 3.0),
+                  (0.0, 5.0, 10.0, 15.0, 20, 25.0, 35.0, 36.0),
+                  (0.0, 5.0, 10.0, 15.0, 20, 25.0, 35.0, 36.0, 47)]
+
+        rng = np.random.default_rng(1234)
+
+        num_trailing_dims = 2
+        trailing_points = [3 + i for i in range(num_trailing_dims)]
```

Comment:
This makes the last dimension have length 4, equal to the `len(sample)`. Would be nice to help future readers of this to have all dimensions different. No need to hold this PR for it, will fix in a follow-up PR.

Reply:
The fix is in https://github.com/scipy/scipy/pull/16829